tardis/tests/tests_slow/test_w7.py

```diff
@@ -102,22 +104,56 @@ def test_luminosity_inner(self):
                 self.obtained_radial1d_model.luminosity_inner)
 
     def test_spectrum(self):
-        assert_quantity_allclose(
+        try:
+            assert_quantity_allclose(
                 self.reference['luminosity_density_nu'],
                 self.obtained_radial1d_model.spectrum.luminosity_density_nu)
 
-        assert_quantity_allclose(
+            assert_quantity_allclose(
                 self.reference['delta_frequency'],
                 self.obtained_radial1d_model.spectrum.delta_frequency)
 
-        assert_quantity_allclose(
+            assert_quantity_allclose(
                 self.reference['wavelength'],
                 self.obtained_radial1d_model.spectrum.wavelength)
 
-        assert_quantity_allclose(
+            assert_quantity_allclose(
                 self.reference['luminosity_density_lambda'],
                 self.obtained_radial1d_model.spectrum.luminosity_density_lambda)
 
+            self.plot_spectrum(passed=True)
+        except Exception as e:
+            self.plot_spectrum(passed=False)
+            raise e
+
+    def plot_spectrum(self, passed):
+        plt.suptitle("Deviation in spectrum_quantities", fontweight="bold")
+
+        # `ldl_` prefixed variables associated with `luminosity_density_lambda`.
+        # Axes of subplot are extracted, if we wish to make multiple plots
+        # for different spectrum quantities all in one figure.
+        ldl_ax = plt.subplot(111)
+        ldl_ax.set_title("Deviation in luminosity_density_lambda")
+        ldl_ax.set_xlabel("N-th packet")
```

Comment:
You are plotting the luminosity density which is used together with the wavelength to form the spectrum.
labeling the x axis N-th packet is wrong.
Plot the data as (x,y) where x is the wavelength and y the luminosity density at that x and call the x-label Wavelength.